File: Sequence Valley/Prefix Sum/Q1. Find the Highest Altitude/Q1_FindtheHighestAltitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int currentAltitude = 0; // Starting altitude
        int maxAltitude = 0; // To keep track of the highest altitude

        // Iterate through the gain array
        for (int i = 0; i < gain.length; i++) {
            currentAltitude += gain[i]; // Update current altitude
            maxAltitude = Math.max(maxAltitude, currentAltitude); // Update max altitude if current is higher
        }

        return maxAltitude; // Return the highest altitude
    }
}

File: Sequence Valley/Prefix Sum/Q1. Find the Highest Altitude/Q1_FindtheHighestAltitude.py
class Solution(object):
    def largestAltitude(self, gain):
        """
        :type gain: List[int]
        :rtype: int
        """
        # Start at altitude 0
        current_altitude = 0
        highest_altitude = 0
        
        # Process each gain to calculate altitude at each point
        for altitude_gain in gain:
            # Add gain to current altitude
            current_altitude += altitude_gain
            
            # Update highest altitude if current is higher
            highest_altitude = max(highest_altitude, current_altitude)
        
        # Return the highest altitude encountered
        return highest_altitude